File: go/00_demarrer/premier_programme.go
/*
 * MODULE 00 - Premier programme en Go
 * ===================================
 * Ton tout premier programme Go. Objectif : voir le cycle
 * « écrire -> lancer » en action (Go compile tout seul juste avant).
 *
 * Pour le lancer, ouvre un terminal et tape :
 *     go run go/00_demarrer/premier_programme.go
 *
 * Rappel : les commentaires (comme ce bloc) sont ignorés par le compilateur.
 * En Go on écrit  // pour une ligne   ou   ... pour plusieurs lignes.
 */

// (1) package main = "ceci est un programme exécutable" (et non une bibliothèque).
// Tout programme Go que l'on peut lancer commence par cette ligne.
package main

// (2) On importe la boîte à outils d'affichage "fmt" (format).
//     Elle contient Println(). C'est l'équivalent d'un import en Python.
import "fmt"

// (3) func main = le POINT DE DÉPART de tout programme Go. L'exécution commence ici.
// L'accolade { ouvre le corps de la fonction (fermé par } tout en bas).
func main() {

	// (4) fmt.Println affiche du texte SUIVI d'un retour à la ligne automatique
	//     (Println = "print line"). Pas besoin de \n à la fin : c'est inclus.
	fmt.Println("Bonjour le monde !")
	fmt.Println("Je viens de lancer mon premier programme Go.")

	// Le programme s'exécute de haut en bas : cette ligne s'affiche en dernier.
	fmt.Println("Cette ligne s'affiche apres les precedentes.")
}

/*
 * À TOI DE MODIFIER :
 *   - change le texte des Println,
 *   - ajoute un nouveau fmt.Println avec ton message,
 *   - relance avec  go run go/00_demarrer/premier_programme.go
 * Avec go run, pas besoin de recompiler à la main : Go le fait automatiquement.
 */

File: java/00_demarrer/PremierProgramme.java
/*
 * MODULE 00 - Premier programme en Java
 * =====================================
 * Le traditionnel "Hello world", commenté pour comprendre CHAQUE morceau.
 *
 * Compiler puis lancer :
 *     javac java/00_demarrer/PremierProgramme.java
 *     java -cp java/00_demarrer PremierProgramme
 */

// (1) En Java, TOUT le code vit à l'intérieur d'une CLASSE.
//     'public' = visible de partout. 'class' = on déclare une classe.
//     ⚠️ Une classe publique DOIT être dans un fichier du MÊME NOM :
//        la classe PremierProgramme est donc dans PremierProgramme.java.
public class PremierProgramme {

    // (2) 'main' est le POINT DE DÉPART : la JVM exécute ce qui est ici en premier.
    //     - public : accessible depuis l'extérieur (la JVM doit pouvoir l'appeler).
    //     - static : appartient à la classe, pas besoin de créer un objet pour l'appeler.
    //     - void   : cette méthode ne renvoie aucune valeur.
    //     - String[] args : les éventuels arguments passés au programme (vide ici).
    public static void main(String[] args) {

        // (3) System.out.println(...) affiche le texte PUIS un retour à la ligne.
        //     Le texte va entre guillemets doubles. L'instruction finit par un ;
        System.out.println("Bonjour, Java !");

        // On peut afficher plusieurs lignes : chaque println écrit sur sa propre ligne.
        System.out.println("Ce code a ete compile en bytecode...");
        System.out.println("...puis execute par la JVM.");

        /* print (sans 'ln') affiche SANS retour à la ligne :
           les deux morceaux ci-dessous restent donc sur la MÊME ligne. */
        System.out.print("Ecris une fois, ");
        System.out.println("execute partout.");
    }
}
